File: src/shared/types/Group.ts
export interface Group {
  id: string;
  courseId: string;
  name: string;
  professorId?: string;
  schedule?: string;
  days?: string[];
  startTime?: string;
  endTime?: string;
  startDate?: string;
  endDate?: string;
  maxStudents: number;
  currentStudents?: number;
  status: "open" | "closed" | "completed";
  createdAt: string;
  updatedAt: string;
  // Include course info for price lookup
  coursePrice?: number;
  courseName?: string;
  // New fields for class duration and skipped dates
  classDuration?: number;
  skippedDates?: string[];
  calculatedEndDate?: string;
}

export interface CreateGroupInput {
  courseId: string;
  name: string;
  professorId?: string;
  schedule?: string;
  days?: string[];
  startTime?: string;
  endTime?: string;
  startDate?: string;
  endDate?: string;
  maxStudents: number;
  classDuration?: number;
  skippedDates?: string[];
}

export interface UpdateGroupInput {
  name?: string;
  schedule?: string;
  days?: string[];
  startTime?: string;
  endTime?: string;
  startDate?: string;
  endDate?: string;
  maxStudents?: number;
  professorId?: string;
  status?: Group["status"];
  classDuration?: number;
  skippedDates?: string[];
}

File: src-tauri/src/application/dto/group.rs
//! Group DTOs

use serde::{Deserialize, Serialize};

/// Create group request
#[derive(Debug, Deserialize)]
pub struct CreateGroupRequest {
    pub course_id: String,
    pub name: String,
    pub professor_id: Option<String>,
    pub schedule: Option<String>,
    pub days: Option<Vec<String>>,
    pub start_time: Option<String>,
    pub end_time: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub max_students: i32,
    pub class_duration: Option<i32>,
    pub skipped_dates: Option<Vec<String>>,
}

/// Update group request
#[derive(Debug, Deserialize)]
pub struct UpdateGroupRequest {
    pub name: Option<String>,
    pub professor_id: Option<String>,
    pub schedule: Option<String>,
    pub days: Option<Vec<String>>,
    pub start_time: Option<String>,
    pub end_time: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub max_students: Option<i32>,
    pub status: Option<String>,
    pub class_duration: Option<i32>,
    pub skipped_dates: Option<Vec<String>>,
}

/// Group DTO
#[derive(Debug, Serialize)]
pub struct GroupDto {
    pub id: String,
    pub course_id: String,
    pub name: String,
    pub professor_id: String,
    pub schedule: Option<String>,
    pub days: Option<Vec<String>>,
    pub start_time: Option<String>,
    pub end_time: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub max_students: i32,
    pub current_students: i32,
    pub status: String,
    pub class_duration: Option<i32>,
    pub skipped_dates: Vec<String>,
    pub calculated_end_date: Option<String>,
}
